src/internal/proc.js

```diff
@@ -356,7 +356,8 @@ export default function proc(
   }
 
   function resolveIterator(iterator, effectId, name, cb) {
-    proc(iterator, subscribe, dispatch, getState, options, effectId, name, cb)
+    const task = proc(iterator, subscribe, dispatch, getState, options, effectId, name, cb)
```

Comment:
Note that here we pass `cb` as last param to `proc`. It will be called once the iterator is done